packages/api/src/pins.js

```diff
@@ -0,0 +1,86 @@
+import { JSONResponse } from './utils/json-response.js'
+import { normalizeCid } from './utils/normalize-cid.js'
+
+/**
+ * @param {import('./user').AuthenticatedRequest} request
+ * @param {import('./env').Env} env
+ * @param {import('./index').Ctx} ctx
+ */
+export async function pinPost (request, env, ctx) {
+  const { cid, name, origins, meta } = await request.json()
+
+  if (!cid) {
+    return new JSONResponse(
+      { error: { reason: `Invalid request id: ${cid}` }},
+      { status: 400 }
+    )
+  }
+
+  try {
+    const normalizedCid = normalizeCid(cid)
+  } catch (err) {
+    return new JSONResponse(err)
```

Comment:
IINW the error format here is slightly different than the one in the spec

```{ error: { reason: string, [detail]: string }}```

It would be great to be consistent with the errors we send back!
We could event create a type to make ourselves accountable to it ?

Also I suggest we keep the reason to be either a textual description associated with the numeric status code or we come up with a set of predefined values to indicate the error in the context of the api.
In both cases we should add more context of what actually happened in the detail.